specification/network/resource-manager/Microsoft.Network/stable/2020-06-01/expressRouteProviderPort.json

```diff
@@ -0,0 +1,170 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "NetworkManagementClient",
+    "description": "The Microsoft Azure Network management API provides a RESTful set of web services that interact with Microsoft Azure Networks service to manage your network resources. The API has entities that capture the relationship between an end user and the Microsoft Azure Networks service.",
+    "version": "2020-06-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network/expressRouteProviderPorts": {
```

Comment:
if there is a LIST API their should be a way to GET an individual expressRouteProviderPorts by name as well

Reply:
customer would not know the name, they would only location name.. should we still implement a get by name?

Reply:
Yes, it is part of the Resource Provider Contract.  If you are exposing a resource with an ID there should be a GET API on that ID to retrieve the resource.  https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md